### PR TITLE
DOC: update submodule docstrings, remove info.py files

### PR DESCRIPTION
**Pull request by rgommers**
Most submodule docstrings are unused for generating the html docs, only for viewing with `?` in IPython or similar. So a short and up-to-date statement of where to go look for info or how to use the module is much more useful than the outdated stuff in the `info.py` files that was there.

### CODE REVIEW

**Comment by eric-wieser** [comment]
These info modules are consumed elsewhere in numpy via a runtime import in some wierd doc function, I think. We might need to remove that too.

**Comment by rgommers** [comment]
> These info modules are consumed elsewhere in numpy via a runtime import in some wierd doc function, I think. We might need to remove that too.

The only ones that come to mind are `np.lookfor`, `np.info` or `np.source`. Those all still seem to work, and I cannot find a trace of anything trying to import `info` dynamically there.

**Comment by rgommers** [comment]
@mattip this does not need a release note, there's nothing user-facing here.

**Comment by eric-wieser** [comment]
I was thinking of `PackageLoader`, which [I already removed](https://github.com/numpy/numpy/pull/11703/files#diff-c527ac8d36c7953549910e594499513cL52)

File: numpy/distutils/__init__.py [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+
+For configuring the preference for and location of libraries like BLAS and
+LAPACK, and for setting include paths and similar build options, please see
+``site.cfg.example`` in the root of the NumPy repository or sdist.
+
+"""
+
 from __future__ import division, absolute_import, print_function
 
-from .__version__ import version as __version__
```
Comment (eric-wieser) [inline numpy/distutils/__init__.py]: The removal of this probably warrants a release note

**Comment by eric-wieser** [comment]
I'd be inclined to add a release note about the `info` modules vanishing, just in case someone is pulling docs from them.

**Comment by rgommers** [comment]
I'm not in favor of adding any release notes here. A couple of reasons:

1. Our release notes are already way too long. They are that long that they exceed the maximum length of what I can put into the Tidelift UI for example, which forces me to edit them down at release time. Also, we want people to _read_ the release notes. Making them super long with stuff that was private or broken to begin with and no one is using is counter-productive.

2. We have tons of cleanup like this to do, and being forced to add release notes for everything would significantly increase the overhead. Example, removing `numpy.math` would be really quick, quicker than adding a release note for it. And `numpy.math` is less unlikely to be in use than `info.py` or `distutils.__version__`.

**Comment by mattip** [comment]
Thanks @rgommers

**Pull request merged**